.github/workflows/ci.yaml

```diff
@@ -0,0 +1,60 @@
+name: Run tests
+
+on:
+  pull_request:
+  push:
+    branches:
+     - main
+     - feat/**
+  schedule:
+    - cron: "53 0 * * *" # Daily at 00:53 UTC
+  workflow_dispatch:
+  workflow_call:
+
```

Comment:
Because I haven't done this yet, I'm keeping this for the final part as there are some decisions to make here on how we store the libs.
I made a bold decision for now to store it in `single_kernel_mongo/lib` but this might change according to the decisions made in this PR.